tools/bench/_timing.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python
+
+import csv
+import time
+
+
+def timeit(lst):
+    def timing_val(func):
+        def wrapper(*args, **kwargs):
+            t1 = time.time()
+            result = func(*args, **kwargs)
+            t2 = time.time()
+            lst.append((func.__name__, (t2 - t1)))
+            return result
+
+        return wrapper
+
+    return timing_val
+
+
+def write(fname, lst, prefix=None):
```

Comment:
```suggestion
def write(
    fname: str,
    timings: <timing type>,
    prefix: str | None = None,
):
    """Appends timing data to the file.
    
    Args:
        fname: The name of the file.
        timings: The timings data to append to the file, one row per timing.
            This list is cleared at the end.
        prefix: An optional prefix for each timing line written.
    """
```

lots of non-obvious parts here that could use documentation:

* This *appends* to a file rather than creating a new one
* `lst` contains timings in the format `(str, float)` (or any value that can be passed to `list(...)`)
* This *clears* the timings list at the end!